src/plugins/template/backend/ops/ctc_loss.cpp

```diff
@@ -2,62 +2,59 @@
 #include "openvino/reference/ctc_loss.hpp"
-// clang-format on
+
+#include "evaluate_node.hpp"
 
 namespace ctc_loss_v4 {
-template <ngraph::element::Type_t t1,
-          ngraph::element::Type_t t2,
-          typename std::enable_if<
-              !std::is_floating_point<typename ngraph::element_type_traits<t1>::value_type>::value &&
-                  !std::is_same<typename ngraph::element_type_traits<t1>::value_type, ngraph::bfloat16>::value &&
-                  !std::is_same<typename ngraph::element_type_traits<t1>::value_type, ngraph::float16>::value,
-              bool>::type = true>
-inline void evaluate(const std::shared_ptr<ngraph::op::v4::CTCLoss>& op,
-                     const ngraph::HostTensorVector& outputs,
-                     const ngraph::HostTensorVector& inputs) {
-    OPENVINO_ASSERT(false,
-                    "The data type for logits is expected to be a floating point type. Got:",
-                    ngraph::element::Type(t1));
+template <
+    ov::element::Type_t t1,
+    ov::element::Type_t t2,
+    typename std::enable_if<!std::is_floating_point<typename ov::element_type_traits<t1>::value_type>::value &&
+                                !std::is_same<typename ov::element_type_traits<t1>::value_type, ov::bfloat16>::value &&
+                                !std::is_same<typename ov::element_type_traits<t1>::value_type, ov::float16>::value,
+                            bool>::type = true>
+inline void evaluate(const std::shared_ptr<ov::op::v4::CTCLoss>& op,
+                     ov::TensorVector& outputs,
+                     const ov::TensorVector& inputs) {
+    OPENVINO_THROW("The data type for logits is expected to be a floating point type. Got:", ov::element::Type(t1));
 }
 
-template <ngraph::element::Type_t t1,
-          ngraph::element::Type_t t2,
-          typename std::enable_if<
-              std::is_floating_point<typename ngraph::element_type_traits<t1>::value_type>::value ||
-                  std::is_same<typename ngraph::element_type_traits<t1>::value_type, ngraph::bfloat16>::value ||
-                  std::is_same<typename ngraph::element_type_traits<t1>::value_type, ngraph::float16>::value,
-              bool>::type = true>
-inline void evaluate(const std::shared_ptr<ngraph::op::v4::CTCLoss>& op,
-                     const ngraph::HostTensorVector& outputs,
-                     const ngraph::HostTensorVector& inputs) {
-    using T1 = typename ngraph::element_type_traits<t1>::value_type;
-    using T2 = typename ngraph::element_type_traits<t2>::value_type;
-    ov::reference::CTCLoss<T1, T2>(inputs[0]->get_data_ptr<T1>(),
-                                   inputs[0]->get_shape(),
-                                   inputs[1]->get_data_ptr<T2>(),
-                                   inputs[2]->get_data_ptr<T2>(),
-                                   inputs[3]->get_data_ptr<T2>(),
-                                   inputs[4]->get_data_ptr<T2>(),
+template <
+    ov::element::Type_t t1,
```

Comment:
It is detail but maybe use ET0, ET1 etc. here?

Reply:
Yes will change in next PR if CI is passed.

Reply:
Fixed